homeassistant/helpers/service.py

```diff
@@ -783,3 +783,43 @@ async def check_permissions(call: ServiceCall) -> Any:
         return check_permissions
 
     return decorator
+
+
+class ReloadServiceHelper:
+    """Helper for reload services to minimize unnecessary reloads."""
+
+    def __init__(self, service_func: Callable[[ServiceCall], Awaitable]):
```

Comment:
This only supports coroutine functions. Is that an acceptable limitation? Do we need to check the type of the passed function in the constructor?

Reply:
That's fine.